tests/screens/test_sync.py

```diff
@@ -375,5 +379,113 @@ async def test_deactivated_jira_user(app, git_repository, helpers, mocker):
 
         assert app.github.load_global_config(app.repo.global_config_source) == {
             'jira_server': 'https://foo.atlassian.net',
-            'members': {},
+            'members': {'bar1': 'jira-bar1', 'foo1': 'jira-foo1'},
+        }
+
+
+async def test_github_user_not_in_jira(app, git_repository, helpers, mocker):
+    app.configure(
+        git_repository,
+        caching=True,
+        data={'github': {'user': 'foo', 'token': 'bar'}, 'jira': {'email': 'foo@bar.baz', 'token': 'bar'}},
+    )
+    mocker.patch(
```

Comment:
I have some concerns with this patching approach. 

IIUC this mocks responses from different services, which means it assumes that the code is going to make a specific set of requests in a very specific order. This ties the test to implementation details that are not related to the behavior being tested. It also makes the test hard to understand for someone that's not familiar with all that context and assumptions.

I think we should ideally mock our own wrappers around those calls rather than `httpx`, but short of that, we may also select the mock response to return based on the URL of the request or something like that. At the bare minimum add some comments about the context and what each response is for.

Reply:
You're right and I was lazy, I copy/pasted the tests that already exist and modified it for my use case.

I pushed a quick fix to also validate the mocks and the arguments they took. I know it's not perfect and we are still too tightly coupled to what's the code is doing under the hood, which the tests should not really care about. But at least we verify more stuff. I'll open a follow-up PR to modify all the tests at once if that's OK for you.